### DISCUSSION TITLE
Using FFMPEG on Gokrazy

### DISCUSSION THREAD

**Discussion by BrunoTeixeira1996** [gokrazy/gokrazy#278] [Q&A] [answered]
Hi guys!
I am currently using a LXC container on Proxmox that only has ffmpeg installed to connect back to a TAPO camera I have.
The ffmpeg is used to record the camera when I am out of home (using a go program).
Is there any option to have this setup on gokrazy? This would be awesome since I could use ffmpeg without having a LXC just for that.

**Comment by psanford** [comment]
I use a statically compiled version of ffmpeg with my gokrazy devices. It works well for me

**Reply by BrunoTeixeira1996** [reply]
Hey, thanks for your feedback!
Is there a way you could resume what you do go get that statically compiled version of ffmpeg + setup on gokrazy?

EDIT: I also use ffprobe to validate my camera, does that also gets compiled?

Thank you!

**Reply by mpl** [reply]
Hey @BrunoTeixeira1996 ,
I'll have to do that as well at some point (using ffprobe to check the validity of some video files). Did you manage to get it statically compiled, or did you go with the podman solution?

**Reply by BrunoTeixeira1996** [reply]
Hi @mpl ,
Neither of those solutions, I had to use LXC on proxmox and run there my ffmpeg program.
However at that time I tested the ffmpeg and ffprobe inside gokrazy and everything worked just fine. I downloaded the binary, copied into gokrazy and then on my go program I used the full path for the binary and I got the same result.

**Comment by stapelberg** [comment, marked as answer]
Aside from statically compiling ffmpeg (see https://gokrazy.org/development/non-go/ for more details), you could also run ffmpeg in a Docker container, see https://gokrazy.org/packages/docker-containers/.

I am running [Frigate](https://frigate.video/), which is a very capable camera system with object/person detection, like so:

```
/usr/local/bin/podman run \
--rm \
--name=frigate \
--shm-size=256m \
--privileged --volume /dev/bus/usb:/dev/bus/usb --env=TZ=Europe/Zurich \
--volume=/perm/frigate-config:/config:rw \
--volume=/perm/nobackup/frigate-media:/media/frigate:rw \
--network=host \
-t \
--entrypoint /bin/sh \
ghcr.io/blakeblackshear/frigate:stable -c 'sed -i "s,listen 5000,listen localhost:5000,g" /usr/local/nginx/conf/nginx.conf && exec /init'
```

**Comment by mpl** [comment]
For future reference, there's even static builds of ffmpeg nicely provided at https://johnvansickle.com/ffmpeg/
I've just tested the arm64 ffprobe one on my gokrazy instance, no problem.
